### PR TITLE
WT-5620 Fix ignorning TOMBSTONE/Stop time pair for HS

### PR DESCRIPTION
**Pull request by kommiharibabu**
Ignoring TOMBSTONE/Stop time pair during search for every scenario
has visibility problems in corner cases. Fix it to ignore only
during rollback to stable.

Transaction id is not required to restore during rollback to stable
as there is no snapshot requirement because of no parallel transactions.

### CODE REVIEW

File: src/txn/txn_rollback_to_stable.c [hunk 1]
```diff
@@ -867,7 +863,11 @@ __wt_rollback_to_stable(WT_SESSION_IMPL *session, const char *cfg[])
      * concurrently.
      */
     WT_RET(__wt_open_internal_session(S2C(session), "txn rollback_to_stable", true, 0, &session));
+
+    F_SET(session, WT_SESSION_RTS);
```
Comment (quchenhao) [inline src/txn/txn_rollback_to_stable.c]: roll back to stable is single threaded? It is out of scope, can we parallelise that for different tables?

Comment (kommiharibabu) [reply src/txn/txn_rollback_to_stable.c]: Yes, rollback to stable is currently single-threaded. Maybe in future, we can parallelize it if we are finding the performance problems with a single thread.

File: src/include/session.h [hunk 2]
```diff
@@ -197,8 +197,9 @@ struct __wt_session_impl {
 #define WT_SESSION_QUIET_CORRUPT_FILE 0x01000000u
 #define WT_SESSION_READ_WONT_NEED 0x02000000u
 #define WT_SESSION_RESOLVING_TXN 0x04000000u
-#define WT_SESSION_SCHEMA_TXN 0x08000000u
-#define WT_SESSION_SERVER_ASYNC 0x10000000u
+#define WT_SESSION_RTS 0x08000000u
```
Comment (agorrod) [inline src/include/session.h]: Is RTS rollback to stable? That's not obvious from the context here.

I feel like we need an abbreviation, but this doesn't seem like the right place to introduce one.

File: src/include/txn.i [hunk 3]
```diff
@@ -795,8 +795,9 @@ __wt_txn_read_upd_list(WT_SESSION_IMPL *session, WT_UPDATE *upd, WT_UPDATE **upd
             continue;
         upd_visible = __wt_txn_upd_visible_type(session, upd);
         if (upd_visible == WT_VISIBLE_TRUE) {
-            /* Don't consider TOMBSTONE updates for the history store. */
-            if (type == WT_UPDATE_TOMBSTONE && WT_IS_HS(S2BT(session)))
+            /* Don't consider TOMBSTONE updates for the history store during rollback to stable. */
```
Comment (agorrod) [inline src/include/txn.i]: I'd rather not capitalize TOMBSTONE in the comment here - comments are supposed to be full English sentences that preferrably don't reference variable names.

Comment (kommiharibabu) [reply src/include/txn.i]: Fixed it with `tombstone`.

File: src/include/session.h (hunk 2 above)
Comment (kommiharibabu) [reply src/include/session.h]: Rename the macro to `WT_SESSION_ROLLBACK_TO_STABLE` for better understanding.

**Pull request merged**